### PR TITLE
Scan group protocols should only allow scan modes set in a scan group as option

### PR DESCRIPTION
**Pull request by attilasuto**
#1055

### CODE REVIEW

File: src/client/South/ConfigurePoints.jsx [hunk 1]
```diff
@@ -171,7 +171,7 @@ const ConfigurePoints = () => {
     const name = `points.${index}.${key}`
     return (
       /* eslint-disable-next-line react/jsx-props-no-spreading */
-      { name, value: <Control onChange={onChange} name={name} {...rest} /> }
+      { name, value: <Control onChange={onChange} name={name} point {...rest} /> }
```
Comment (burgerni10) [inline src/client/South/ConfigurePoints.jsx:174]: I'm not sure to understand the 'point' props : in OIbScanMode component, it is marked as a boolean but here it seems to me it is an object.

File: src/client/components/OIbForm/OIbScanMode.jsx [hunk 2]
```diff
@@ -1,12 +1,19 @@
 import React from 'react'
 import PropTypes from 'prop-types'
+import { useParams } from 'react-router-dom'
 import { FormGroup, FormText, Label, Input, FormFeedback } from 'reactstrap'
 import { ConfigContext } from '../../context/configContext.jsx'
 
-const OIbScanMode = ({ label, help, valid, value, name, onChange }) => {
+const OIbScanMode = ({ label, help, valid, value, name, onChange, point }) => {
```
Comment (burgerni10) [inline src/client/components/OIbForm/OIbScanMode.jsx:7]: See my previous comment on the point type (file ConfigurePoint)

File: src/client/components/OIbForm/OIbScanMode.jsx [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
   const { newConfig } = React.useContext(ConfigContext)
   const scanModes = newConfig?.engine?.scanModes ?? [] // scan modes defined in engine
-  const options = scanModes.map((e) => e.scanMode)
+  const { dataSourceId } = useParams()
+  const dataSource = newConfig?.south.dataSources.find((d) => d.dataSourceId === dataSourceId)
+  const scanGroups = dataSource ? dataSource[dataSource.protocol]?.scanGroups : null
+  // Scan Group protocols should only allow scan modes set in a scan group as option
+  const options = scanGroups !== undefined && point
```
Comment (burgerni10) [inline src/client/components/OIbForm/OIbScanMode.jsx:14]: My guess is that you want to display scanGroups only in point list window. Maybe this props should have another name to better understand its role.
Please enlighten me if I misunderstand this props

**Review by burgerni10 — CHANGES_REQUESTED** [review]

File: src/client/components/OIbForm/OIbScanMode.jsx (hunk 3 above)
Comment (attilasuto) [reply src/client/components/OIbForm/OIbScanMode.jsx:14]: Yes, exactly this is only needed when configuring a new point. Maybe we can use another naming like: isPointConfig.

Comment (attilasuto) [reply src/client/components/OIbForm/OIbScanMode.jsx]: I have removed the point flag.

**Review by burgerni10 — APPROVED** [review]

**Pull request merged**